File: Python/Problem_003.py
def largest_prime_factor(n: int) -> int:
    max_factor = 1

    while n % 2 == 0:
        max_factor = 2
        n //= 2

    i = 3
    while i * i <= n:
        while n % i == 0:
            max_factor = i
            n //= i
        i += 2

    if n > 1:
        max_factor = n

    return max_factor


if __name__ == "__main__":
    print(largest_prime_factor(600851475143))  # 6857

File: Java/Problem_003.java
public class Main {
    public static void main(String[] args) {
        long n = 600851475143L;
        long maxFactor = 1;

        while (n % 2 == 0) {
            maxFactor = 2;
            n /= 2;
        }

        for (long i = 3; i * i <= n; i += 2) {
            while (n % i == 0) {
                maxFactor = i;
                n /= i;
            }
        }

        if (n > 1)
            maxFactor = n;

        System.out.println(maxFactor); // 6857
    }
}
